### PR TITLE
Added linters to verify actions.yaml and config.yaml (CRAFT-554).

### LINKED ISSUE #547: `charmcraft analyze` - check for invalid `config.yaml`
Follow on from [canonical/operator#618](https://github.com/canonical/operator/pull/618). 

At present, it is possible to successfully `charmcraft pack` on a charm with an empty `config.yaml`, i.e. one that does not contain an `options:` map. Given that Juju will refuse to deploy a charm with such a `config.yaml`, we should probably try to catch this using the `charmcraft analyze` machinery that is invoked when a charm is packed.

As I understand, you can completely omit `config.yaml`, but if it exists it must have a top level key named `options`.

### PR DESCRIPTION
**Pull request by facundobatista**
Fixes #547 and #310.

### CODE REVIEW

**Comment by facundobatista** [comment]
FTR, these two new linters were added to the [documentation](https://discourse.charmhub.io/t/analyzers-and-linters/4796).

File: charmcraft/linters.py [hunk 1]
```diff
@@ -233,10 +235,86 @@ def run(self, basedir: pathlib.Path) -> str:
         return result
 
 
+class JujuActions:
+    """Check that the actions.yaml file is valid YAML if it exists."""
+
+    check_type = CheckType.lint
+    name = "juju-actions"
+    url = "https://juju.is/docs/sdk/charmcraft-analyze#heading--juju-actions"
```
Comment (cmatsuoka) [inline charmcraft/linters.py:243]: Fixed a typo and slightly rephrased the linked text.

File: charmcraft/linters.py [hunk 2]
```diff
@@ -233,10 +235,86 @@ def run(self, basedir: pathlib.Path) -> str:
...
+    The file is considered valid if the following checks are true:
+
+    - has an 'options' key
+    - it is a dictionary
+    - each item inside has the mandatory 'type' key
+    """
+
+    check_type = CheckType.lint
+    name = "juju-config"
+    url = "https://juju.is/docs/sdk/charmcraft-analyze#heading--juju-config"
```
Comment (cmatsuoka) [inline charmcraft/linters.py:277]: Slightly rephrased the linked text to fix a typo.

**Review by cmatsuoka — APPROVED** [review]
The verifications match the valid file description.

File: charmcraft/linters.py (hunk 2 above)
Comment (facundobatista) [reply charmcraft/linters.py:277]: thanks!

**Pull request merged**